core/client/templates/settings/general.hbs

```diff
@@ -28,9 +28,9 @@
         </fieldset>
 
         <div class="form-group">
-            <label for="blog-logo">Blog Logo</label>
+            <label>Blog Logo</label>
             {{#if model.logo}}
-                <button type="button" class="js-modal-logo" {{action "openModal" "upload" this "logo"}}><img id="blog-logo" {{bind-attr src=model.logo}} alt="logo"></button>
+                <img class="blog-logo" {{bind-attr src=model.logo}} alt="logo" {{action "openModal" "upload" this "logo"}}>
```

Comment:
This seems like an accessibility degradation. At the least, the alt should be updated - I'm sure things can be done with aria/roles, but I'm fairly ignorant in that department